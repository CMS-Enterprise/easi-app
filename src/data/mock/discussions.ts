import {
  SystemIntakeGRBReviewDiscussionFragment,
  SystemIntakeGRBReviewerRole,
  SystemIntakeGRBReviewerVotingRole
} from 'gql/gen/graphql';

const mockDiscussions = (
  systemIntakeID: string = '0d191c27-4de3-46d8-9ded-c2e9c6b3fbd6'
): SystemIntakeGRBReviewDiscussionFragment[] => [
  {
    __typename: 'SystemIntakeGRBReviewDiscussion',
    initialPost: {
      __typename: 'SystemIntakeGRBReviewDiscussionPost',
      id: '882357e4-c0b0-44ef-b749-f71879ad7878',
      content:
        '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>',
      votingRole: SystemIntakeGRBReviewerVotingRole.VOTING,
      grbRole: SystemIntakeGRBReviewerRole.SUBJECT_MATTER_EXPERT,
      createdByUserAccount: {
        __typename: 'UserAccount',
        id: '034fa2b3-00ff-4ec6-857e-75291a59df74',
        commonName: 'Adeline Aarons'
      },
      systemIntakeID,
      createdAt: '2024-11-12T21:59:07.368862Z'
    },
    replies: [
      {
        __typename: 'SystemIntakeGRBReviewDiscussionPost',
        id: '4099dbb7-2752-4bf9-a3e1-da225ceb9fae',
        content:
          '<p>Nisi nobis consectetur voluptatem neque tempore. Ea nihil sed beatae?</p>',
        votingRole: SystemIntakeGRBReviewerVotingRole.NON_VOTING,
        grbRole: SystemIntakeGRBReviewerRole.CMCS_REP,
        createdByUserAccount: {
          __typename: 'UserAccount',
          id: '909a7888-4d6f-4bbf-9b9d-71ec1e2c3068',
          commonName: 'Jerry Seinfeld'
        },
        systemIntakeID,
        createdAt: '2024-11-13T21:59:07.368862Z'
      },
      {
        __typename: 'SystemIntakeGRBReviewDiscussionPost',
        id: '47b0081d-de33-4514-b68f-a7e2bbf2610f',
        content:
          '<p>Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?</p>',
        votingRole: SystemIntakeGRBReviewerVotingRole.VOTING,
        grbRole: SystemIntakeGRBReviewerRole.CO_CHAIR_CIO,
        createdByUserAccount: {
          __typename: 'UserAccount',
          id: '601d52be-7baa-4b45-91cd-88b4a5935c3f',
          commonName: 'Cosmo Kramer'
        },
        systemIntakeID,
        createdAt: '2024-11-13T21:59:07.368862Z'
      }
    ]
  }
];

export default mockDiscussions;
