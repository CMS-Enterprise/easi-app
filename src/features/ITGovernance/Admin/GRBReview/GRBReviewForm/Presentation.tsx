import React from 'react';

import { EasiFormProvider, useEasiForm } from 'components/EasiForm';

import GRBReviewFormStepWrapper from './GRBReviewFormStepWrapper';

// TODO: Update fields type
type PresentationFields = {};

const Presentation = () => {
  const form = useEasiForm<PresentationFields>();

  return (
    <EasiFormProvider<PresentationFields> {...form}>
      <GRBReviewFormStepWrapper>
        <p>Fields here</p>
      </GRBReviewFormStepWrapper>
    </EasiFormProvider>
  );
};

export default Presentation;
