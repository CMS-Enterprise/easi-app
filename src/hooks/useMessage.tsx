import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

import Alert, { AlertProps } from 'components/shared/Alert';

type MessageProps = {
  message?: string | React.ReactElement;
  alertProps?: Omit<AlertProps, 'children'>;
};

type SetMessage = (
  message?: string | React.ReactElement,
  alertProps?: Omit<AlertProps, 'children'>
) => void;

const MessageContext = createContext<
  | {
      Message: ({ className }: { className?: string }) => JSX.Element | null;
      showMessage: SetMessage;
      showMessageOnNextPage: SetMessage;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<MessageProps>({});
  const [message, setMessage] = useState<MessageProps>({});

  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  const Message = (props: { className?: string }) => {
    // If no message, return null
    if (!message.message) return null;

    // If message has alert props, wrap in component
    if (message.alertProps) {
      return (
        <Alert {...message.alertProps} {...props}>
          {message.message}
        </Alert>
      );
    }

    // Return plain message
    return <>{message.message}</>;
  };

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage({});
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        Message,
        showMessage: (value, alertProps) =>
          setMessage({
            message: value,
            alertProps
          }),
        showMessageOnNextPage: (value, alertProps) =>
          setQueuedMessage({
            message: value,
            alertProps
          })
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
