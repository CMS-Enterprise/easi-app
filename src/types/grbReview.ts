import {
  GetGRBReviewersComparisonsQuery,
  SystemIntakeGRBReviewerRole,
  SystemIntakeGRBReviewerVotingRole
} from 'gql/gen/graphql';

export type GRBReviewerFields = {
  userAccount: {
    username: string;
    commonName: string;
    email: string;
  };
  votingRole: SystemIntakeGRBReviewerVotingRole;
  grbRole: SystemIntakeGRBReviewerRole;
};

export type GRBReviewerComparisonIntake =
  GetGRBReviewersComparisonsQuery['compareGRBReviewersByIntakeID'];

export type GRBReviewerComparison =
  GRBReviewerComparisonIntake[number]['reviewers'][number];
