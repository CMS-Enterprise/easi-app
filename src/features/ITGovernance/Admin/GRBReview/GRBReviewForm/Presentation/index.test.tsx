import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { grbReview } from 'tests/mock/grbReview';

import { MessageProvider } from 'hooks/useMessage';

import Presentation from '.';

describe('GRB review form - presentation', () => {
  it('renders the step', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          `/it-governance/${grbReview.id}/grb-review/presentation`
        ]}
      >
        <MessageProvider>
          <Route path="/it-governance/:systemId/grb-review/:step">
            <Presentation grbReview={grbReview} />
          </Route>
        </MessageProvider>
      </MemoryRouter>
    );

    expect(
      screen.getByRole('heading', {
        name: 'Step 2 of 4 Presentation'
      })
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
