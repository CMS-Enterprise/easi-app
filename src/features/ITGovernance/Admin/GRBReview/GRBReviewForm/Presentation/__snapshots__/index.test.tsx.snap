// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GRB review form - presentation > renders the step 1`] = `
<DocumentFragment>
  <div>
    <div
      class="trb-form-header"
    >
      <div
        class="bg-gray-5 padding-bottom-6"
      >
        <div
          class="grid-container"
          data-testid="gridContainer"
        >
          <div
            class="grid-row"
            data-testid="grid"
          >
            <div
              class="grid-col"
              data-testid="grid"
            >
              <nav
                aria-label="Breadcrumbs"
                class="usa-breadcrumb usa-breadcrumb--wrap padding-bottom-0"
                style="background-color: transparent;"
              >
                <ol
                  class="usa-breadcrumb__list"
                >
                  <li
                    class="usa-breadcrumb__list-item"
                  >
                    <a
                      class="text-primary usa-breadcrumb__link"
                      href="/"
                    >
                      <span>
                        Home
                      </span>
                    </a>
                  </li>
                  <li
                    class="usa-breadcrumb__list-item"
                  >
                    <a
                      class="text-primary usa-breadcrumb__link"
                      href="/it-governance/a4158ad8-1236-4a55-9ad5-7e15a5d49de2/grb-review"
                    >
                      <span>
                        IT Governance request details
                      </span>
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="usa-breadcrumb__list-item usa-current"
                  >
                    <span>
                      Set up GRB review
                    </span>
                  </li>
                </ol>
              </nav>
              <h1
                aria-live="polite"
                class="easi-h1 margin-bottom-0"
                tabindex="-1"
              >
                Set up GRB review
              </h1>
              <div
                class="font-body-lg line-height-body-5 text-light"
              >
                Use this form to set up a Governance Review Board (GRB) review in EASi.
              </div>
              <div
                class="margin-top-2 line-height-body-5"
              >
                You will be able to select the type of review, add recordings and documents, invite GRB members to review this project, and set a timeframe for an asynchronous review.
              </div>
              <div
                class="margin-top-3"
              >
                <button
                  class="usa-button usa-button--unstyled display-flex flex-align-center"
                  data-testid="button"
                  type="button"
                >
                  <svg
                    class="usa-icon margin-right-1"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                  Save and return to request
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid-container"
        data-testid="gridContainer"
      />
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <div
          aria-label="progress"
          class="usa-step-indicator margin-top-4 margin-bottom-2"
          data-testid="step-indicator"
        >
          <ol
            class="usa-step-indicator__segments"
          >
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-0"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Review type 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
            <li
              aria-current="true"
              aria-disabled="true"
              class="usa-step-indicator__segment usa-step-indicator__segment--current maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-1"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Presentation 
              </span>
            </li>
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-2"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Additional documentation 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-3"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Participants and timeframe 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
          </ol>
          <div
            class="usa-step-indicator__header"
          >
            <h4
              class="usa-step-indicator__heading"
            >
              <span
                class="usa-step-indicator__heading-counter"
              >
                <span
                  class="usa-sr-only"
                >
                  Step
                </span>
                <span
                  class="usa-step-indicator__current-step"
                >
                  2
                </span>
                 
                <span
                  class="usa-step-indicator__total-steps"
                >
                  of 4
                </span>
                 
              </span>
              <span
                class="usa-step-indicator__heading-text"
              >
                Presentation
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid-container grb-review-form-presentation padding-bottom-10"
      data-testid="gridContainer"
    >
      <form
        class="usa-form step-content-wrapper maxw-none"
        data-testid="grbReviewForm-stepContentWrapper"
        role="form"
      >
        <p
          class="line-height-body-5 text-light font-body-sm margin-top-0 margin-bottom-1"
        >
          Organize a date and time for the project team to record or present their project. For asynchronous reviews, also add links to the presentation materials.
        </p>
        <p
          class="text-base margin-top-0 font-body-sm"
        >
          Fields marked with an asterisk (
          <span
            class="text-red margin-x-05"
          >
            *
          </span>
          ) are required.
        </p>
        <fieldset
          class="usa-fieldset"
          data-testid="fieldset"
        >
          <div
            class="desktop:grid-col-6"
            data-testid="grid"
          >
            <div
              class="border-top-1px border-base-lighter margin-top-3"
            >
              <h4
                class="margin-top-1 margin-bottom-05"
              >
                GRB Meeting
              </h4>
              <p
                class="margin-y-0 text-base"
              >
                Organize a GRB meeting for the project team’s presentation to the GRB.
              </p>
              <div
                class="usa-alert usa-alert--info usa-alert--slim margin-top-105"
                data-testid="alert"
              >
                <div
                  class="usa-alert__body"
                >
                  <p
                    class="usa-alert__text"
                  >
                    You will need to work with the project team outside of EASi to set up a calendar invite, but you may include the date below for reference purposes within EASi.
                  </p>
                </div>
              </div>
              <div
                class="usa-form-group margin-top-5"
                data-testid="formGroup"
              >
                <div
                  class="usa-form-group"
                  data-testid="formGroup"
                >
                  <label
                    class="usa-label text-normal"
                    data-testid="label"
                    for="grbMeetingDate.grbDate"
                  >
                    GRB meeting date 
                    <abbr
                      class="usa-hint usa-hint--required"
                      title="required"
                    >
                      *
                    </abbr>
                  </label>
                  <div
                    class="easi-help-text margin-top-1"
                  >
                    This date will be visible to the requester. Format: mm/dd/yyyy
                  </div>
                  <div
                    class="usa-date-picker usa-date-picker--initialized"
                    data-testid="date-picker"
                  >
                    <input
                      aria-hidden="true"
                      class="usa-input usa-sr-only usa-date-picker__internal-input"
                      data-testid="date-picker-internal-input"
                      name="grbMeetingDate.grbDate"
                      readonly=""
                      tabindex="-1"
                      type="text"
                      value=""
                    />
                    <div
                      class="usa-date-picker__wrapper"
                      tabindex="-1"
                    >
                      <input
                        class="usa-input usa-date-picker__external-input"
                        data-testid="date-picker-external-input"
                        id="grbMeetingDate.grbDate"
                        type="text"
                        value=""
                      />
                      <button
                        aria-haspopup="true"
                        aria-label="Toggle calendar"
                        class="usa-date-picker__button"
                        data-testid="date-picker-button"
                        type="button"
                      />
                      <div
                        class="usa-date-picker__calendar"
                        data-testid="date-picker-calendar"
                        hidden=""
                        role="application"
                        style="top: 0px;"
                      />
                      <div
                        aria-live="polite"
                        class="usa-sr-only usa-date-picker__status"
                        data-testid="date-picker-status"
                        role="status"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="usa-form-group margin-top-6"
                data-testid="formGroup"
              >
                <li
                  class="usa-card maxw-none margin-bottom-0"
                  data-testid="Card"
                >
                  <div
                    class="usa-card__container margin-0 shadow-2 radius-md"
                  >
                    <header
                      class="usa-card__header"
                      data-testid="CardHeader"
                    >
                      <h4
                        class="margin-bottom-1"
                      >
                        Presentation deck
                      </h4>
                      <p
                        class="margin-top-0 text-base"
                      >
                        The requester and their project team are responsible for uploading their completed presentation deck to EASi, though admin team members may upload a deck for them.
                      </p>
                    </header>
                    <div
                      class="usa-card__body"
                      data-testid="CardBody"
                    >
                      <div
                        class="usa-alert usa-alert--info usa-alert--slim"
                        data-testid="alert"
                      >
                        <div
                          class="usa-alert__body"
                        >
                          <p
                            class="usa-alert__text"
                          >
                            The requester has not yet uploaded their GRB presentation deck. Use the buttons below to send a reminder or upload the deck yourself.
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="usa-card__footer"
                      data-testid="CardFooter"
                    >
                      <button
                        class="usa-button margin-top-0 margin-right-1"
                        data-testid="button"
                        type="button"
                      >
                        Send reminder
                      </button>
                      <label
                        for="presentationDeck.presentationDeckFileData"
                      >
                        <button
                          class="usa-button usa-button--unstyled margin-top-1"
                          data-testid="button"
                          type="button"
                        >
                          or, upload a deck for the project team
                        </button>
                      </label>
                      <input
                        accept=".pdf,.doc,.docx,.ppt,.pptx,.xls,.xlsx"
                        aria-describedby="Select your file"
                        data-testid="file-upload-input"
                        id="presentationDeck.presentationDeckFileData"
                        name="presentationDeck.presentationDeckFileData"
                        style="display: none;"
                        type="file"
                      />
                    </div>
                  </div>
                </li>
              </div>
            </div>
          </div>
        </fieldset>
        <div
          class="margin-top-8 margin-bottom-3"
        >
          <ul
            class="usa-button-group"
          >
            <li
              class="usa-button-group__item"
            >
              <div
                class="display-flex flex-align-center"
              >
                <button
                  class="usa-button usa-button--outline margin-top-0 margin-bottom-1 mobile-lg:margin-bottom-0"
                  data-testid="button"
                  type="button"
                >
                  Back
                </button>
              </div>
            </li>
            <li
              class="usa-button-group__item"
            >
              <div
                class="display-flex flex-align-center"
              >
                <button
                  class="usa-button margin-top-0"
                  data-testid="button"
                  disabled=""
                  type="submit"
                >
                  Next
                </button>
              </div>
            </li>
          </ul>
          <button
            class="usa-button usa-button--unstyled margin-top-2 display-flex flex-align-center"
            data-testid="button"
            type="button"
          >
            <svg
              class="usa-icon margin-right-05"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
            Save and return to request
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GRB review form - presentation > renders the step and takes a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="trb-form-header"
    >
      <div
        class="bg-gray-5 padding-bottom-6"
      >
        <div
          class="grid-container"
          data-testid="gridContainer"
        >
          <div
            class="grid-row"
            data-testid="grid"
          >
            <div
              class="grid-col"
              data-testid="grid"
            >
              <nav
                aria-label="Breadcrumbs"
                class="usa-breadcrumb usa-breadcrumb--wrap padding-bottom-0"
                style="background-color: transparent;"
              >
                <ol
                  class="usa-breadcrumb__list"
                >
                  <li
                    class="usa-breadcrumb__list-item"
                  >
                    <a
                      class="text-primary usa-breadcrumb__link"
                      href="/"
                    >
                      <span>
                        Home
                      </span>
                    </a>
                  </li>
                  <li
                    class="usa-breadcrumb__list-item"
                  >
                    <a
                      class="text-primary usa-breadcrumb__link"
                      href="/it-governance/a4158ad8-1236-4a55-9ad5-7e15a5d49de2/grb-review"
                    >
                      <span>
                        IT Governance request details
                      </span>
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="usa-breadcrumb__list-item usa-current"
                  >
                    <span>
                      Set up GRB review
                    </span>
                  </li>
                </ol>
              </nav>
              <h1
                aria-live="polite"
                class="easi-h1 margin-bottom-0"
                tabindex="-1"
              >
                Set up GRB review
              </h1>
              <div
                class="font-body-lg line-height-body-5 text-light"
              >
                Use this form to set up a Governance Review Board (GRB) review in EASi.
              </div>
              <div
                class="margin-top-2 line-height-body-5"
              >
                You will be able to select the type of review, add recordings and documents, invite GRB members to review this project, and set a timeframe for an asynchronous review.
              </div>
              <div
                class="margin-top-3"
              >
                <button
                  class="usa-button usa-button--unstyled display-flex flex-align-center"
                  data-testid="button"
                  type="button"
                >
                  <svg
                    class="usa-icon margin-right-1"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                  Save and return to request
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid-container"
        data-testid="gridContainer"
      />
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <div
          aria-label="progress"
          class="usa-step-indicator margin-top-4 margin-bottom-2"
          data-testid="step-indicator"
        >
          <ol
            class="usa-step-indicator__segments"
          >
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-0"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Review type 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
            <li
              aria-current="true"
              aria-disabled="true"
              class="usa-step-indicator__segment usa-step-indicator__segment--current maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-1"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Presentation 
              </span>
            </li>
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-2"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Additional documentation 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-3"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Participants and timeframe 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
          </ol>
          <div
            class="usa-step-indicator__header"
          >
            <h4
              class="usa-step-indicator__heading"
            >
              <span
                class="usa-step-indicator__heading-counter"
              >
                <span
                  class="usa-sr-only"
                >
                  Step
                </span>
                <span
                  class="usa-step-indicator__current-step"
                >
                  2
                </span>
                 
                <span
                  class="usa-step-indicator__total-steps"
                >
                  of 4
                </span>
                 
              </span>
              <span
                class="usa-step-indicator__heading-text"
              >
                Presentation
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid-container grb-review-form-presentation padding-bottom-10"
      data-testid="gridContainer"
    >
      <form
        class="usa-form step-content-wrapper maxw-none"
        data-testid="grbReviewForm-stepContentWrapper"
        role="form"
      >
        <p
          class="line-height-body-5 text-light font-body-sm margin-top-0 margin-bottom-1"
        >
          Organize a date and time for the project team to record or present their project. For asynchronous reviews, also add links to the presentation materials.
        </p>
        <p
          class="text-base margin-top-0 font-body-sm"
        >
          Fields marked with an asterisk (
          <span
            class="text-red margin-x-05"
          >
            *
          </span>
          ) are required.
        </p>
        <fieldset
          class="usa-fieldset"
          data-testid="fieldset"
        >
          <div
            class="desktop:grid-col-6"
            data-testid="grid"
          >
            <div
              class="border-top-1px border-base-lighter margin-top-3"
            >
              <h4
                class="margin-top-1 margin-bottom-05"
              >
                GRB Meeting
              </h4>
              <p
                class="margin-y-0 text-base"
              >
                Organize a GRB meeting for the project team’s presentation to the GRB.
              </p>
              <div
                class="usa-alert usa-alert--info usa-alert--slim margin-top-105"
                data-testid="alert"
              >
                <div
                  class="usa-alert__body"
                >
                  <p
                    class="usa-alert__text"
                  >
                    You will need to work with the project team outside of EASi to set up a calendar invite, but you may include the date below for reference purposes within EASi.
                  </p>
                </div>
              </div>
              <div
                class="usa-form-group margin-top-5"
                data-testid="formGroup"
              >
                <div
                  class="usa-form-group"
                  data-testid="formGroup"
                >
                  <label
                    class="usa-label text-normal"
                    data-testid="label"
                    for="grbMeetingDate.grbDate"
                  >
                    GRB meeting date 
                    <abbr
                      class="usa-hint usa-hint--required"
                      title="required"
                    >
                      *
                    </abbr>
                  </label>
                  <div
                    class="easi-help-text margin-top-1"
                  >
                    This date will be visible to the requester. Format: mm/dd/yyyy
                  </div>
                  <div
                    class="usa-date-picker usa-date-picker--initialized"
                    data-testid="date-picker"
                  >
                    <input
                      aria-hidden="true"
                      class="usa-input usa-sr-only usa-date-picker__internal-input"
                      data-testid="date-picker-internal-input"
                      name="grbMeetingDate.grbDate"
                      readonly=""
                      tabindex="-1"
                      type="text"
                      value=""
                    />
                    <div
                      class="usa-date-picker__wrapper"
                      tabindex="-1"
                    >
                      <input
                        class="usa-input usa-date-picker__external-input"
                        data-testid="date-picker-external-input"
                        id="grbMeetingDate.grbDate"
                        type="text"
                        value=""
                      />
                      <button
                        aria-haspopup="true"
                        aria-label="Toggle calendar"
                        class="usa-date-picker__button"
                        data-testid="date-picker-button"
                        type="button"
                      />
                      <div
                        class="usa-date-picker__calendar"
                        data-testid="date-picker-calendar"
                        hidden=""
                        role="application"
                        style="top: 0px;"
                      />
                      <div
                        aria-live="polite"
                        class="usa-sr-only usa-date-picker__status"
                        data-testid="date-picker-status"
                        role="status"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="usa-form-group margin-top-6"
                data-testid="formGroup"
              >
                <li
                  class="usa-card maxw-none margin-bottom-0"
                  data-testid="Card"
                >
                  <div
                    class="usa-card__container margin-0 shadow-2 radius-md"
                  >
                    <header
                      class="usa-card__header"
                      data-testid="CardHeader"
                    >
                      <h4
                        class="margin-bottom-1"
                      >
                        Presentation deck
                      </h4>
                      <p
                        class="margin-top-0 text-base"
                      >
                        The requester and their project team are responsible for uploading their completed presentation deck to EASi, though admin team members may upload a deck for them.
                      </p>
                    </header>
                    <div
                      class="usa-card__body"
                      data-testid="CardBody"
                    >
                      <div
                        class="usa-alert usa-alert--info usa-alert--slim"
                        data-testid="alert"
                      >
                        <div
                          class="usa-alert__body"
                        >
                          <p
                            class="usa-alert__text"
                          >
                            The requester has not yet uploaded their GRB presentation deck. Use the buttons below to send a reminder or upload the deck yourself.
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="usa-card__footer"
                      data-testid="CardFooter"
                    >
                      <button
                        class="usa-button margin-top-0 margin-right-1"
                        data-testid="button"
                        type="button"
                      >
                        Send reminder
                      </button>
                      <label
                        for="presentationDeck.presentationDeckFileData"
                      >
                        <button
                          class="usa-button usa-button--unstyled margin-top-1"
                          data-testid="button"
                          type="button"
                        >
                          or, upload a deck for the project team
                        </button>
                      </label>
                      <input
                        accept=".pdf,.doc,.docx,.ppt,.pptx,.xls,.xlsx"
                        aria-describedby="Select your file"
                        data-testid="file-upload-input"
                        id="presentationDeck.presentationDeckFileData"
                        name="presentationDeck.presentationDeckFileData"
                        style="display: none;"
                        type="file"
                      />
                    </div>
                  </div>
                </li>
              </div>
            </div>
          </div>
        </fieldset>
        <div
          class="margin-top-8 margin-bottom-3"
        >
          <ul
            class="usa-button-group"
          >
            <li
              class="usa-button-group__item"
            >
              <div
                class="display-flex flex-align-center"
              >
                <button
                  class="usa-button usa-button--outline margin-top-0 margin-bottom-1 mobile-lg:margin-bottom-0"
                  data-testid="button"
                  type="button"
                >
                  Back
                </button>
              </div>
            </li>
            <li
              class="usa-button-group__item"
            >
              <div
                class="display-flex flex-align-center"
              >
                <button
                  class="usa-button margin-top-0"
                  data-testid="button"
                  disabled=""
                  type="submit"
                >
                  Next
                </button>
              </div>
            </li>
          </ul>
          <button
            class="usa-button usa-button--unstyled margin-top-2 display-flex flex-align-center"
            data-testid="button"
            type="button"
          >
            <svg
              class="usa-icon margin-right-05"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
            Save and return to request
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
